Hide Zombie Logging

maxyeg opened the pull request:
For review.
Not sure about the names. 
And I can't pass 'nginx-bogus-logging' option to beaker(maybe because of windows).

Intended to fix Issue #3550

Commit:
Hide Zombie Logging

scottdraves commented on core/src/main/java/com/twosigma/beaker/core/Main.java:
```diff
@@ -96,6 +96,7 @@ private static CommandLine parseCommandLine(String[] args)
     opts.addOption(null, "require-password", false, "Ask for password when connecting");
     opts.addOption(null, "listen-interface", true, "Interface to listen on - requires ip address or '*'");
     opts.addOption(null, "portable", false, "Configuration and runtime files located in application instead of user home directory.");
+    opts.addOption(null, "nginx-bogus-logging", false, "Switchs off logging unsuccessfull requests from outdated beaker pages.");
```
nginx-bogus-logging -> hide-zombie-logging

default should be true.

Switchs off logging unsuccessfull requests from outdated beaker pages.
->
Prevent distracting logging by Beaker clients of previous server instances.

Commit:
#3550. Change parameter names

Commit:
#3550. Zombie logging.
Rename variables.
Change regexp.
Mark plugins' rules with '^~' to stop searching for other rules.

scottdraves commented on core/src/main/java/com/twosigma/beaker/core/rest/PluginServiceLocatorRest.java:
```diff
@@ -727,6 +737,12 @@ else if (nginxRule.equals("ipython2"))
+          this.CATCH_OUTDATED_REQUESTS_RULE);
+    } else {
+      nginxConfig = nginxConfig.replace("%(catch_outdated_requests_rule)s", "");
+    }
```
better style:

nginxConfig = nginxConfig.replace("%(catch_outdated_requests_rule)s", this.hideZombieLogging ? this.CATCH_OUTDATED_REQUESTS_RULE : "");

Commit:
#3550. Zombie logging.
Invert logic for this option.
Make inline condition.

scottdraves committed:
fix doc to match option function

scottdraves committed:
merge
